riscv-c-api.md

```diff
@@ -116,5 +116,41 @@ For example:
 
 ## Intrinsic Functions
 
-Do we really have none of these?  I can't figure out
-`gcc/gcc/config/riscv/riscv-builtins.c`...
+RISC-V intrinsics allow access to RISC-V-specific instructions from C or C++ that are usually not otherwise synthesizable by compilers.
+Intrinsics typically map to a single instruction that is inlined to avoid function call overhead.
+
+Intrinsics are only available if the corresponding instructions are available (as part of the effective `-march` string).
+Some intrinsics are only available if a particular header file is included.
+RISC-V header files, that enable intrinsics, require the prefix `riscv_` (e.g. `riscv_vector.h` or `riscv_crypto.h`).
+
+RISC-V intrinsics follow the following naming rule:
+
+```
+INTRINSIC ::= PREFIX '_' MNEMONIC [ '_' RET_TYPE ]
+PREFIX ::= "__builtin_riscv" | "__rv" EXTENSION | "_rv" | "_rv32" | "_rv64"
```

Comment:
- `__builtin_riscv`: I would suggest drop "__builtin_riscv", that might give people some impression is that could be used without include header file.

- `"__rv" EXTENSION`: I would suggest `__rv` and no `EXTENSION` name encoded there (or just remove that to use `_rv`?), there are some instruction are appeared in different extension like `rol` and `ror`? Although they defined with `_rv32` and `_rv64` in crypto intrinsic only, but I would like to prevent we have something like `__rvX_zz` and `__rvY_zz` with same semantic but different intrinsic name in future.